### PR TITLE
WIP fix redis cli for validate_script_output('redis-cli get foo', sub { m/(nil)/ })

### PR DESCRIPTION
**Pull request by Zaoliang**
need to use svirt remote worker on OSD

[type description here, PLEASE, REMOVE THIS LINE, PLACEHOLDER, BEFORE SUBMITTING THIS PULL REQUEST]

- Related ticket: https://progress.opensuse.org/issues/xyz
- Needles: https://github.com/os-autoinst/os-autoinst-needles-opensuse/pull/xyz
- Verification run: openqa.mypersonalinstance.de/tests/xyz#step/module/x

### CODE REVIEW

File: schedule/functional/redis.yaml [hunk 1]
```diff
@@ -0,0 +1,17 @@
+---
```
Comment (pdostal) [inline schedule/functional/redis.yaml:1]: This is not necessary. Just use `INCLUDE_MODULES=bootloader_start,boot_to_desktop,redis` when you're running your verifications ;)

File: tests/console/redis.pm [hunk 2]
```diff
@@ -37,7 +38,7 @@ sub run {
     validate_script_output('redis-cli get foo', sub { m/bar/ });
     validate_script_output('redis-cli pfselftest', sub { m/OK/ });
     validate_script_output('redis-cli flushdb', sub { m/OK/ });
-    validate_script_output('redis-cli get foo', sub { m/(nil)/ });
+    validate_script_output('redis-cli get foo', sub { m/(nil)/ }) if (!is_svirt);
```
Comment (pdostal) [inline tests/console/redis.pm:41]: This is not needed. The output is just not `nil` but `\n` so it can be done better with some perl magic :smile:

Comment (pdostal) [reply tests/console/redis.pm:41]: Maybe this can help: https://perlmaven.com/check-if-string-is-empty-or-has-only-spaces-in-perl

**Review by pdostal — CHANGES_REQUESTED** [review]

File: schedule/functional/redis.yaml (hunk 1 above)
Comment (Zaoliang) [reply schedule/functional/redis.yaml:1]: yes, thanks!

File: tests/console/redis.pm (hunk 2 above)
Comment (Zaoliang) [reply tests/console/redis.pm:41]: I still don't understand this, this is working on	qemu_x86_64. Do you mean to use sub { m/(nil) \n/ } ?
thanks!

File: tests/console/redis.pm [hunk 3]
```diff
@@ -37,7 +37,7 @@ sub run {
     validate_script_output('redis-cli get foo', sub { m/bar/ });
     validate_script_output('redis-cli pfselftest', sub { m/OK/ });
     validate_script_output('redis-cli flushdb', sub { m/OK/ });
-    validate_script_output('redis-cli get foo', sub { m/(nil)/ });
+    validate_script_output('redis-cli get foo', sub { m/(nil\n)/ });
```
Comment (foursixnine) [inline tests/console/redis.pm:40]:
```suggestion
    validate_script_output('redis-cli get foo', sub { chomp; m/(nil)/ });
```

Comment (Zaoliang) [reply tests/console/redis.pm]: https://openqa.suse.de/tests/9635022#step/redis/21 still not working, thanks!

**Comment by foursixnine** [comment]
@Zaoliang you can close this one now. Happens that flushdb was deleting all keys, so just checking that the key didn't exist in the store, was good enough

**Comment by Zaoliang** [comment]
close this pr because of #15630, thanks for your feedback and input!

File: tests/console/redis.pm (hunk 2 above)
Comment (pdostal) [reply tests/console/redis.pm]: Something like that. In qemu backend it is `(null)` but in `svirt` it's probably `\n`

**Pull request closed without merging**